plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSource.java

```diff
@@ -245,6 +251,16 @@ public long getMemoryUsage()
         return memoryContext.getBytes();
     }
 
+    @Override
+    public Metrics getMetrics()
+    {
+        CompressionKind compressionKind = recordReader.getCompressionKind();
```

Comment:
Instead of getting this via recordReader, I think you can just pass it into OrcPageSource constructor from OrcReader